src/screens/Connectors/ConnectorMetaData/BankDebit/BankDebit.res

```diff
@@ -0,0 +1,181 @@
+    />
+  }
+}
+
+@react.component
+let make = (
+  ~onCloseClickCustomFun,
+  ~setShowWalletConfigurationModal,
+  ~update,
+  ~paymentMethod,
+  ~paymentMethodType,
+  ~initialValues,
+  ~setInitialValues,
+) => {
+  open LogicUtils
+  open BankDebitUtils
+  let connectorList = HyperswitchAtom.connectorListAtom->Recoil.useRecoilValueFromAtom
+  let formState: ReactFinalForm.formState = ReactFinalForm.useFormState(
+    ReactFinalForm.useFormSubscription(["values"])->Nullable.make,
+  )
+  let form = ReactFinalForm.useForm()
+
+  let connectorsListPMAuth =
+    connectorList->ConnectorUtils.getProcessorsListFromJson(
+      ~removeFromList=ConnectorTypes.PMAuthenticationProcessor,
+    )
+  let pmAuthConnectors =
+    connectorsListPMAuth->Array.map(item => item.connector_name)->removeDuplicate
+
+  let options = pmAuthConnectors->dropdownOptions
+
+  let getConnectorId = connector => {
+    let connectorData = connectorsListPMAuth->Array.find(item => {
+      item.connector_name == connector
+    })
+    switch connectorData {
+    | Some(connectorData) => connectorData.merchant_connector_id
+    | None => ""
+    }
+  }
+
+  let onCancelClick = () => {
+    let existingPaymentMethodValues =
+      formState.values
+      ->getDictFromJsonObject
+      ->getDictfromDict("pm_auth_config")
+      ->getArrayFromDict("enabled_payment_methods", [])
+      ->JSON.Encode.array
+      ->getArrayDataFromJson(itemToObjMapper)
+
+    let newPaymentMethodValues =
+      existingPaymentMethodValues->Array.filter(item =>
+        item.payment_method_type !== paymentMethodType
+      )
+
+    form.change(
+      "pm_auth_config.enabled_payment_methods",
+      newPaymentMethodValues->Identity.genericTypeToJson,
+    )
+  }
+
+  let closeModal = () => {
+    onCloseClickCustomFun()
+    onCancelClick()
+    setShowWalletConfigurationModal(_ => false)
+  }
+
+  let onSubmit = () => {
+    setShowWalletConfigurationModal(_ => false)
+    setInitialValues(_ => formState.values)
+    update()
+    Nullable.null->Promise.resolve
+  }
+
+  let renderValueInp = (options: array<SelectBox.dropdownOption>) => (
+    fieldsArray: array<ReactFinalForm.fieldRenderProps>,
+  ) => {
+    <PmtConfigInp options fieldsArray paymentMethod paymentMethodType getConnectorId />
+  }
+
+  let valueInput = (inputArg: PaymentMethodConfigTypes.valueInput) => {
+    open FormRenderer
+    makeMultiInputFieldInfoOld(
+      ~label=`${inputArg.label}`,
+      ~comboCustomInput=renderValueInp(inputArg.options),
+      ~inputFields=[makeInputFieldInfo(~name=`${inputArg.name1}`), makeInputFieldInfo(~name=``)],
+      (),
+    )
+  }
+
+  <>
+    <div className="p-4 border border-red-300">
+      <FormRenderer.FieldRenderer
+        field={valueInput({
+          name1: `pm_auth_config.enabled_payment_methods`,
+          name2: ``,
+          label: `Select the open banking verification provider to verify the bank accounts`,
+          options,
```

Comment:
Can we clearly define what options refer to, rather than using the generic term 'options'?

Reply:
options are the dropdown options object that we want to show in the dropdown, it is a prop of the fieldRenderer component

Reply:
Can we name the variable based on the context of the dropdown values, such as connectorsOptions or paymentMethodOptions, instead of using a generic name like options?